backend/src/test/java/de/propertyfinder/backend/service/PropertyServiceTest.java

```diff
@@ -0,0 +1,188 @@
+import de.propertyfinder.backend.model.AdditionalPurchaseCosts;
+import de.propertyfinder.backend.model.PlzGermany;
+import de.propertyfinder.backend.model.Property;
+import de.propertyfinder.backend.repo.AdditionalPurchaseCostRepo;
+import de.propertyfinder.backend.repo.PlzRepo;
+import de.propertyfinder.backend.repo.PropertyRepo;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.*;
+
+class PropertyServiceTest {
+
+    private final PlzRepo plzRepo = mock(PlzRepo.class);
+    private final PropertyRepo propertyRepo = mock(PropertyRepo.class);
+    private final AdditionalPurchaseCostRepo additionalPurchaseCostRepo = mock(AdditionalPurchaseCostRepo.class);
+    private final OctoparseApiService octoparseApiService = mock(OctoparseApiService.class);
+    private final PropertyMapper propertyMapper = mock(PropertyMapper.class);
+
+    private final PropertyService propertyService = new PropertyService(propertyRepo, propertyMapper, octoparseApiService, plzRepo, additionalPurchaseCostRepo);
+
+    @Test
+    @DisplayName("set state depending on plz of property")
+    void setStateTest() {
+        //Given
+        Property property = Property.builder()
+                .plz("78267")
+                .state("")
+                .build();
+
+        PlzGermany plzGermany = PlzGermany.builder()
+                .bundesland("Baden-Württemberg")
+                .build();
+
+        //When
+        when(plzRepo.findByPlz(property.getPlz())).thenReturn(plzGermany);
+        Property actual = propertyService.setState(property);
+
+        //Then
+        Property expected = Property.builder()
+                .state("Baden-Württemberg")
+                .plz("78267")
+                .build();
+        assertThat(actual, is(expected));
+    }
+
+    @Test
+    @DisplayName("set real estate Agent Fee")
+    void setRealEstateAgentFeeTest() {
+        //Given
+        AdditionalPurchaseCosts additionalPurchaseCosts = AdditionalPurchaseCosts.builder()
+                .state("Baden-Württemberg")
+                .realEstateAgentFeeInPercent(0.0357)
+                .build();
+        Property property = Property.builder()
+                .plz("78267")
+                .state("Baden-Württemberg")
+                .purchasePriceInEuro(1000.0)
+                .realEstateAgentFeeInPercent(0.0)
+                .realEstateAgentFeeInEuro(0.0)
+                .build();
+        //When
+        when(additionalPurchaseCostRepo.findByState("Baden-Württemberg")).thenReturn(additionalPurchaseCosts);
+        Property actual = propertyService.setRealEstateAgentFee(property);
+        //Then
+        Property expected = Property.builder()
+                .plz("78267")
+                .state("Baden-Württemberg")
+                .purchasePriceInEuro(1000.0)
+                .realEstateAgentFeeInPercent(0.0357)
```

Comment:
these names are a bit misleading. The Fee you're saving is not actually in percent